nuxeo-core/nuxeo-core-storage/src/main/java/org/nuxeo/ecm/core/storage/DefaultFulltextParser.java

```diff
@@ -61,8 +74,8 @@ public String parse(String s, String path) {
      * This can be subclassed.
      */
     @Override
-    public void parse(String s, String path, List<String> strings) {
-        s = preprocessField(s, path);
+    public void parse(String s, String path, String mimeType, DocumentRef docRef, List<String> strings) {
```

Comment:
`docRef` is not used in the code. If you actually need it in some custom code, please make a mention in the implementation that this parameter is currently unused but has some use cases for potential subclasses.

However a `DocumentRef` is not enough to uniquely determine a document, as you have no way of knowing the repository name. This would be manifest if there was actual code trying to use this parameter. You should use a `DocumentLocation` instead.